apps/admin-gui/src/app/shared/components/dialogs/bulk-invite-members-dialog/bulk-invite-members-dialog.component.ts

```diff
@@ -0,0 +1,170 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  UntypedFormBuilder,
+  UntypedFormControl,
+  ValidatorFn,
+  Validators,
+} from '@angular/forms';
+import { downloadData, emailRegexString } from '@perun-web-apps/perun/utils';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { StoreService } from '@perun-web-apps/perun/services';
+import { InputInvitationsFromCsv, RegistrarManagerService } from '@perun-web-apps/perun/openapi';
+
+export interface BulkInviteMembersDialogData {
+  theme: string;
+  voId: number;
+  groupId: number;
+}
+
+interface OutputData {
+  email: string;
+  status: string;
+}
+
+@Component({
+  selector: 'app-bulk-invite-members-dialog',
+  templateUrl: './bulk-invite-members-dialog.component.html',
+  styleUrls: ['./bulk-invite-members-dialog.component.scss'],
+})
+export class BulkInviteMembersDialogComponent implements OnInit {
+  invitedMembers = new UntypedFormControl('', [Validators.required, this.userInputValidator()]);
+  theme: string;
+  languages = ['en'];
+  currentLanguage = 'en';
+  loading = false;
+  state = 'input';
+  finishedWithErrors: boolean;
+
+  private resultData: { [p: string]: string };
+
+  constructor(
+    public dialogRef: MatDialogRef<BulkInviteMembersDialogComponent>,
+    private formBuilder: UntypedFormBuilder,
+    @Inject(MAT_DIALOG_DATA) public data: BulkInviteMembersDialogData,
+    private store: StoreService,
+    private registrarManager: RegistrarManagerService
+  ) {}
+
+  private static didSomeInviteFail(resultData: { [p: string]: string }): boolean {
+    for (const key in resultData) {
+      if (resultData[key] !== 'OK') {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  ngOnInit(): void {
+    this.theme = this.data.theme;
+    this.languages = this.store.getProperty('supported_languages');
+  }
+
+  onCancel(): void {
+    this.dialogRef.close(false);
+  }
+
+  onSubmit(): void {
+    this.loading = true;
+    const listOfMembers = (this.invitedMembers.value as string).split('\n');
+    const parsedMembers: string[] = [];
+    for (const line of listOfMembers) {
+      const parsedLine = this.parseMemberLine(line);
+      if (parsedLine !== 'format' && parsedLine !== 'email') {
+        if (parsedLine !== '') {
+          parsedMembers.push(parsedLine);
+        }
```

Comment:
Do we even need this check here ? It should not be possible to click on submit when the format is invalid.